proto-lens-protoc/src/Generate.hs

```diff
@@ -266,24 +269,41 @@ generateEnumDecls info =
             ]
         ]
     ]
+    ++
+    -- pattern FooAlias :: Foo
+    -- pattern FooAlias = FooConstructor
+    concat
+        [ [ PatSynSig noLoc aliasName Nothing [] [] dataType
+          , PatSyn noLoc (PVar aliasName) (PVar originalName)
+                ImplicitBidirectional
+          ]
+        | EnumValueInfo
+            { enumValueName = aliasName
+            , enumAliasOf = Just originalName
+            } <- enumValues info
+        ]
   where
     dataType = TyCon $ UnQual dataName
-    EnumInfo
-        { enumName = dataName
-        , enumValueNames = valueNames
-        , enumDescriptor = ed
-        } = info
-    namesAndValues = zip valueNames (ed ^. value)
-    names = map (second (^. name)) namesAndValues
-    values = List.sortBy (comparing snd) $
-        map (second (fromIntegral . (^. number))) namesAndValues
-    minBoundName = fst $ head values
-    maxBoundName = fst $ last values
+    EnumInfo { enumName = dataName, enumDescriptor = ed } = info
+    constructors :: [(Name, EnumValueDescriptorProto)]
+    constructors = List.sortBy (comparing ((^. number) . snd))
+                            [(n, d) | EnumValueInfo
+                                { enumValueName = n
+                                , enumValueDescriptor = d
+                                , enumAliasOf = Nothing
+                                } <- enumValues info
+                            ]
+    constructorNames = map fst constructors
+    minBoundName = head constructorNames
+    maxBoundName = last constructorNames
+
+    constructorProtoNames = map (second (^. name)) constructors
+    constructorNumbers = map (second (fromIntegral . (^. number)))
+                          constructors
 
-    succPairs = zip ctorNames $ tail ctorNames
-      where ctorNames = map fst values
+    succPairs = zip constructorNames $ tail constructorNames
     succDecl funName boundName thePairs = InsDecl $ FunBind $
-        match funName [PApp (UnQual boundName) []] (
+        match funName [PVar boundName] (
```

Comment:
Reverted, sorry for the confusion.

It does work correctly either way by coincidence, but you're right that it should be PApp.